lib/core/src/Cardano/Wallet/CoinSelection/Internal.hs

```diff
@@ -235,14 +235,20 @@ data SelectionError
       SelectionOutputError
     deriving (Eq, Show)
 
+-- TODO: ADP-1448:
+--
+-- Replace this type synonym with a type parameter on types that use it.
+--
+type InputId = (TxIn, Address)
+
 -- | Represents a balanced selection.
 --
 data Selection = Selection
     { inputs
-        :: !(NonEmpty (TxIn, TokenBundle))
+        :: !(NonEmpty (InputId, TokenBundle))
         -- ^ Selected inputs.
     , collateral
-        :: ![(TxIn, Coin)]
+        :: ![(InputId, Coin)]
         -- ^ Selected collateral inputs.
     , outputs
         :: ![(Address, TokenBundle)]
```

Comment:
> There are some remaining `TxOut` and `Address` types... Are you sure we won't end up needing a output type-parameter anyway? Maybe this interplays with the choices in this PR

I do eventually plan to remove all of the `TxOut` references, I promise! I haven't got there yet, as I'm currently trying to do these changes one area at a time, partly in order to limit the size and complexity of each PR. But all of them will go away!

As for the `outputs` field: I suspect this is one area where we might need to use an `output` type parameter, simply because it's helpful to the caller to be able to link the outputs they supplied to the outputs they eventually get back in the final selection.

User-specified outputs can be changed by the algorithm (if they didn't specify any ada, then we allow ourselves to add the minimum ada quantity).

But perhaps there's a way to avoid even this, in future, if we take the responsibility for "preparing outputs" away from coin selection.